server/pulp/server/db/model/__init__.py

```diff
@@ -471,6 +495,19 @@ def get_repositories(self):
         return Repository.objects(repo_id__in=id_list)
 
     @property
+    def downloaded(self):
+        """
+        Whether all of the files associated with the unit have been downloaded.
+
+        :return: True if all downloaded.
+        :rtype: bool
+        """
+        for file in self.files:
```

Comment:
Grrr!  I hate the way python floods the global namespace with builtins.  It takes up so many useful names.  Python should require explicit import like:
```
from builtins import file, list
```